### PR TITLE
Fix editor debounce

### LINKED ISSUE #74: Code Editor sometimes misses input or moves the cursor around
Somehow during validation (?) the editor gets stuck and misses some inputs. 
If its not easily fixable  we should validate everything on blur or make the debounce time longer?

### PR DESCRIPTION
**Pull request by schmanu**
This PR fixes the debounce issues of the CSV Editor by removing the debounce option from the ace-editor component and wrapping the parsing/validation/balance check into a debounced function. 

Furthermore:
Refactors some logic away from the App.tsx into CSVForm.tsx because there were no other dependencies for those functions and states.

fixes #74

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "bignumber.js": "^9.0.1",
     "ethers": "^5.0.24",
     "honeyswap-default-token-list": "^3.1.1",
+    "lodash.debounce": "^4.0.8",
```
Comment (bh2smith) [inline package.json:32]: I thought we were removing this... I guess this is needed to remove it?

**Review by bh2smith — APPROVED** [review]
It looks like a lot of big chunks of logic have been shuffled around. I guess they were moved into places that were more appropriate or necessary. Could you elaborate on the reason for that?

File: package.json (hunk 1 above)
Comment (schmanu) [reply package.json:32]: Its a small subset of the big lodash library just giving us the debounce function which I wanna use to debounce the validation and parsing of the csv.

**Comment by schmanu** [comment]
> It looks like a lot of big chunks of logic have been shuffled around. I guess they were moved into places that were more appropriate or necessary. Could you elaborate on the reason for that?

Most of our logic was in App.tsx, which in my opinion should be more of a small entry point for the app.
This caused the CSVForm Component to have a bunch of properties (changeCallback, submitCallback, csvContent, transfers, ...)
I realized that none of these properties were needed by other components. So it was easy to move it all to CSVForm. I think components should handle as much as possible on their own and not communicate everything all the way to the top level component if not necessary.

But I'm also not an React Expert. It just felt right :man_shrugging:

**Pull request merged**